eureka-core/src/main/java/com/netflix/eureka/EurekaBootStrap.java

```diff
@@ -82,6 +81,12 @@
 
     protected volatile EurekaServerContext serverContext;
     protected volatile AwsBinder awsBinder;
+    
+    private DiscoveryClient discoveryClient;
+    
+    protected void setDiscoveryClient(DiscoveryClient discoveryClient) {
```

Comment:
Rather than providing overrideable ways to set the client, maybe we can add the necessary constructors to EurekaBootstrap:
```
public EurekaBootstrap(ApplicationInfoManager infoManager, EurekaClient client) {
    this... = ...
}

public EurekaBootstrap() {
    this(null, null);
}